addons/sale_timesheet/models/account_move.py

```diff
@@ -10,6 +10,20 @@ class AccountMove(models.Model):
 
     timesheet_ids = fields.One2many('account.analytic.line', 'timesheet_invoice_id', string='Timesheets', readonly=True, copy=False)
     timesheet_count = fields.Integer("Number of timesheets", compute='_compute_timesheet_count')
+    timesheet_encode_uom_id = fields.Many2one('uom.uom', related='company_id.timesheet_encode_uom_id')
+    timesheet_total_duration = fields.Integer("Timesheet Total Duration", compute='_compute_timesheet_total_duration', help="Total recorded duration, expressed in the encoding UoM, and rounded to the unit")
+
+    @api.depends('timesheet_ids', 'company_id.timesheet_encode_uom_id')
+    def _compute_timesheet_total_duration(self):
+        for invoice in self:
+            timesheets = invoice.timesheet_ids if self.user_has_groups('hr_timesheet.group_hr_timesheet_approver') else invoice.timesheet_ids.filtered(lambda t: t.user_id.id == self.env.uid)
+            total_time = 0.0
+            for timesheet in timesheets:
+                # Timesheets may be stored in a different unit of measure, so first we convert all of them to the reference unit
+                total_time += timesheet.unit_amount * timesheet.product_uom_id.factor_inv
+            # Now convert to the proper unit of measure
+            total_time *= invoice.timesheet_encode_uom_id.factor
+            invoice.timesheet_total_duration = round(total_time)
```

Comment:
I do think we should try to improve performance of this compute, even if it's only used in a form view (for now)
I wrote something naive like this : 
```suggestion
    @api.depends('timesheet_ids', 'company_id.timesheet_encode_uom_id')
    def _compute_timesheet_total_duration(self):
        group_data = self.env['account_analytic_line'].read_group([
            ('move_id', 'in', self.ids)
        ], ['move_id', 'unit_amount'], ['move_id'])
        timesheet_unit_amount_dict = defaultdict(float)
        timesheet_unit_amount_dict.update({data['move_id'][0]: data['unit_amount'] for data in group_data})
        for invoice in self:
            total_time = invoice.company_id.project_time_mode_id._compute_quantity(timesheet_unit_amount_dict[invoice.id], invoice.timesheet_encode_uom_id)
            invoice.timesheet_total_duration = round(total_time)
```

And maybe we should try to improve https://github.com/odoo/odoo/blob/3c7626fe0861854d3fb51d9cbef206683935deec/addons%2Fsale_timesheet%2Fmodels%2Fsale_order.py#L43-L53

Don't you think @IT-Ideas ?

Reply:
(Done)